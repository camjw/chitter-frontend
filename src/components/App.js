import React, { Component } from 'react';

export default class App extends Component {
  componentDidMount() {
    this.updatePeepList();
  }

  updatePeepList = () => this.props.getPeeps();
  clearPeepList = () => this.props.clearPeeps();

  render() {
    console.log('PROPS')
    console.log(this.props)
    return (
      <div>
        <h1>Chitter</h1>

        <h2>Last updated at: {this.props.isFetching}</h2>

        <button onClick={() => this.updatePeepList()}>
          Refresh Feed
        </button>

        <button onClick={() => this.clearPeepList()}>
          Clear Feed
        </button>

        {/* <ul>
          {this.props.peeps.map((peep, index) => (
            <li key={index}>
              <p>
                {peep.body} peeped by {peep.user.handle}
              </p>
            </li>
          ))}
        </ul> */}

      </div>
    );
  }
}
