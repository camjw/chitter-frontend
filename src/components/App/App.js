import React from 'react';
import PropTypes from 'prop-types';
import ChitterFeed from '../ChitterFeed';
import UserSignUpContainer from '../../containers/UserSignUpContainer';

export default class App extends React.Component {
  constructor(props) {
    super();
    const { getPeeps } = props;
    this.updatePeepList = () => getPeeps();
  }

  componentDidMount() {
    this.updatePeepList();
  }

  render() {
    const { receivedAt, isFetching, peeps } = this.props;
    return (
      <div className="container">
        <div className="row">
          <div className="col-sm-6">
            <div className="row">
              <h1>Chitter</h1>
            </div>
            <div className="row">
              <UserSignUpContainer />
            </div>
          </div>
          <div className="col-sm-6">
            <div className="row">
              <div className="col-sm">
                {receivedAt
                  ? (
                    <h2>
                  Last updated
                      {' '}
                      {new Date(receivedAt).toLocaleTimeString()}
                    </h2>
                  )
                  : <h2>Loading Chitter Feed</h2>
              }
              </div>
              <div className="col-sm">
                <button className='btn btn-primary' type="submit" onClick={() => this.updatePeepList()}>
                  Refresh Feed
                </button>
              </div>
            </div>
            <div className="row">
              {isFetching ? null
                : <ChitterFeed peeps={peeps} />
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  getPeeps: PropTypes.func.isRequired,
  isFetching: PropTypes.bool.isRequired,
  receivedAt: PropTypes.number,
  peeps: PropTypes.arrayOf(PropTypes.shape()),
};

App.defaultProps = {
  receivedAt: null,
  peeps: [],
};
