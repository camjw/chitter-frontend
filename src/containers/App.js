import App from '../components/App'
import { connect } from 'react-redux';
import { getPeeps, clearPeeps } from '../actions';

const mapStateToProps = (state, ownProps) => ({ peeps: state.peeps });
const mapDispatchToProps = { getPeeps, clearPeeps };
const AppContainer = connect(mapStateToProps, mapDispatchToProps)(App);

export default AppContainer;
