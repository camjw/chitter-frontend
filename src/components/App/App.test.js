import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () => {
  describe('#render', () => {
    it('calls the getPeeps function', () => {
      App.prototype.getPeeps = jest.fn();
      const getPeeps = jest.fn();
      const clearPeeps = jest.fn();
      let appWrapper = shallow(<App getPeeps={getPeeps} clearPeeps={clearPeeps} isFetching={false} />);
      expect(appWrapper.html()).toEqual('hello');
    });
  });
});
