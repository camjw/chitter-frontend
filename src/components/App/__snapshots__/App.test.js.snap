// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App #render informs the user that peeps are being fetched if isFetching is true 1`] = `"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col-sm-4\\"><div class=\\"row\\"><h1>Chitter</h1></div><div class=\\"row\\"><h3> Hello </h3></div></div><div class=\\"col-sm-8\\"><div class=\\"row\\"><div class=\\"col-sm\\"><h2>Loading Chitter Feed</h2></div><div class=\\"col-sm\\"><button type=\\"submit\\">Refresh Feed</button></div></div><div class=\\"row\\"></div></div></div></div>"`;
